arangod/Aql/DocumentProducingHelper.cpp

```diff
@@ -39,35 +45,41 @@ void aql::handleProjections(std::vector<std::string> const& projections,
                             transaction::Methods const* trxPtr, VPackSlice slice,
                             VPackBuilder& b, bool useRawDocumentPointers) {
   for (auto const& it : projections) {
-    if (it == StaticStrings::IdString) {
-      VPackSlice found = transaction::helpers::extractIdFromDocument(slice);
-      if (found.isCustom()) {
-        // _id as a custom type needs special treatment
-        b.add(it, VPackValue(transaction::helpers::extractIdString(trxPtr->resolver(),
-                                                                   found, slice)));
-      } else {
-        b.add(it, found);
-      }
-    } else if (it == StaticStrings::KeyString) {
-      VPackSlice found = transaction::helpers::extractKeyFromDocument(slice);
-      if (useRawDocumentPointers) {
-        b.add(VPackValue(it));
-        b.addExternal(found.begin());
-      } else {
-        b.add(it, found);
+    if (it[0] == '_') {
+      if (it == StaticStrings::IdString) {
+        VPackSlice found = transaction::helpers::extractIdFromDocument(slice);
+        if (found.isCustom()) {
+          // _id as a custom type needs special treatment
+          b.add(it, VPackValue(transaction::helpers::extractIdString(trxPtr->resolver(),
```

Comment:
if you already have the value, can't you just transform it into the string directly ?

Reply:
Fixed